pom.xml

```diff
@@ -88,21 +90,55 @@
         <dependency>
             <groupId>org.scala-lang</groupId>
             <artifactId>scala-library</artifactId>
-            <version>${scala.version}</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.scala-lang</groupId>
+            <artifactId>scala-compiler</artifactId>
+            <optional>true</optional>
+        </dependency>
         <dependency>
             <groupId>org.scalatest</groupId>
             <artifactId>scalatest_${scala.binary.version}</artifactId>
             <version>3.1.0</version>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>org.json4s</groupId>
+            <artifactId>json4s-jackson_${scala.binary.version}</artifactId>
+            <version>${json4s.version}</version>
```

Comment:
All dependencies' versions are hardcoded, excepts for those that are used multiple times for different artifacts from the same group (e.g. scala stuff, json4s stuff). Just a DRY measure.

`dependencyManagement` section only contains Scala artifacts because there was a conflict in transitive dependencies in some optional libraries. I put all Scala artifacts in that section for the sake of completeness. (We could extract them as a BOM and deploy on Sonatype, just like Spring BOM for instance. I googled but didn't find one)